doc/source/tune/key-concepts.rst

```diff
@@ -4,112 +4,117 @@
 Key Concepts
 ============
 
-Let's quickly walk through the key concepts you need to know to use Tune. In this guide, we'll be covering the following:
+.. TODO: should we introduce checkpoints as well?
+.. TODO: should we at least mention "Stopper" classes here?
 
-.. contents::
-    :local:
-    :depth: 1
+Let's quickly walk through the key concepts you need to know to use Tune.
+If you want to see practical tutorials right away, go visit our :ref:`user guides<tune-guides>`.
+In essence, Tune has six crucial components that you need to understand.
 
-.. image:: /images/tune-workflow.png
+First, you define the hyperparameters you want to tune in a `search space` and pass them into a ``Trainable``
+that specifies the objective you want to tune.
+Then you select a `search algorithm` to effectively optimize your parameters and optionally use a
+`scheduler` to stop searches early and speed up your experiments.
+Together with other configuration, your ``Trainable``, algorithm, and scheduler are passed into ``tune.run()``,
+which runs your experiments and creates `trials`.
+These trials can then be used in `analyses` to inspect your experiment results.
```

Comment:
"Trainable" should have the same formatting as the names of the other components IMO

Reply:
@Yard1  The single back ticks emphasise, double back ticks stand for code. Trainable is a class, hence this convention. Then again, there's a certain ambiguity because we use "trainable" very colloquially here as well, e.g. for the function API.

In a way, I feel like "trainable" is a relatively poor convention. We'd generally be better off to call this concept simply an "objective".

Anyway, not sure what's right here.

Reply:
Yeah, I am aware of the distinction between single and double backtics - my issue here is that Trainable is the only name that is being formatted as code, despite the possibility of formatting the other names as well. We should either format everything as classes, or format nothing IMO.

Reply:
understood. addressed that point 👍